### PR TITLE
Fix bgp polling for BGP4-MIB devices

### PR DESCRIPTION
**Pull request by averzicco**
bgp peers polling fails for devices using BGP4-MIB mib (note: discovery works fine) because it tries with CISCO-BGP4-MIB mib.
The fix add some checks on the os_group to apply the cisco mib only for cisco devices and fallback to the generic ones otherwise.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by averzicco** [comment]
somehow tests were failing because of ios_3560g device/variant on bgp-peers module, test for that device uses the generic BGP mib even if it should use the cisco one (at least judging from cisco documentation).

Added test data for bgp-peer module as variant of pfsense with frr as routing software

File: includes/discovery/functions.inc.php [hunk 1]
```diff
@@ -1176,6 +1176,10 @@ function build_bgp_peers($device, $data, $peer2)
             }
         }
         if ($peer && $peer_ip != '0.0.0.0') {
+            if ($peer_as < 0) {
+                //if ASN is negative -> overflow int32 -> original number is max(INT32) - min(INT32) + 1 + value
+                $peer_as = 2147483647 - (-2147483648) + 1 + $peer_as;
```
Comment (murrant) [inline includes/discovery/functions.inc.php:1181]: `$peer_as + 4294967296` ?

**Review by murrant — APPROVED** [review]

**Comment by murrant** [comment]
Thanks for the fixes @averzicco

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-69-release-changelog-november-2020/14124/1

**Pull request merged**